contracts/AelinUpFrontDeal.sol

```diff
@@ -400,41 +410,47 @@ contract AelinUpFrontDeal is AelinERC20, MinimalProxyFactory, IAelinUpFrontDeal
         }
     }
 
+    function claimUnderlyingMutlipleEntries(uint256[] memory _indices) external {
```

Comment:
this is good but maybe we can make our 721 collection enumerable. I am not 100% sure but I am pretty sure that if we add the enumerable extension we can simply use that to retrieve every NFT they own during the claiming process. However, say someone owns 1000 NFTs then enumerating all of them would be very expensive so perhaps passing an array of IDs is the best way. Can you just look into the Enumerable extension and let me know what you think. We dont have to make any changes around this. It is pretty good the way you set it up

Reply:
Yeah I thought about that the other day actually. I will investigate maybe it's worth doing it.

Reply:
Yeah I had a second thought about it and I don't think we need enumerable. I think it's good to keep it quite low level. There is no real friction for the user as an efficient front end will just grab the tokenIDs for you and pass them in the function. And as you said, it could become quite expensive if a user has a lot of tokens. Also they might not want to vest them all at once, we can't really take this decision for them.